### ISSUE TITLE
VanillaFixes

### ISSUE THREAD

**Issue by shikulja** [allfoxwy/add-dll-to-exe#2]
Patching a game is a dirty method.
Some VanillaFixes patches were transferred by SuperWoWhook, some remained.
It is possible to create an additional .dll library into which patches will be transferred. Or add to existing ones (sideload-DLL.dll)

It will also be possible to enable and disable VanillaFixes options from a file, without patching the .exe

I’m writing to the injector topic, because I’m not sure that all the patches can be transferred to .dll

**Comment by allfoxwy** [comment]
Yes, these thing would happen. But loader can't tell what is happening inside a .dll, nor is there any conflict occurring.

Currently we depend on mods play well with others.

**Comment by shikulja** [comment]
Checking the compatibility of library modifications would be useful. 
But I'm just saying that it's a good idea to transfer the remaining VanillaFixes patches that have not yet been added to SuperWoW, and those that may be slightly different

**Comment by allfoxwy** [comment]
Ah, by "transfer" you mean implement same patch in SuperWoW instead of using VanillaFixes?

I'm NOT the author of SuperWoW, I couldn't do that as SuperWoW is closed source.

And personally, I would try implementing something new instead of rewrite something already exist.

VanillaFixes is good, we could continue use it.

**Comment by shikulja** [comment]
yes SuperWoW does not have all the features to get rid of the VanillaFixes patch
Therefore, just a suggestion for the future, create a separate modification library for this.

The most important thing is that the dirty method will not be applied to the client, we will get rid of the patching method that requires other actions from the user,
and make the entire modification more consistent and homogeneous.

**Comment by shikulja** [comment]
Antiviruses also complain about patching with false positives. But the injector has an advantage: we keep the original file untouched, which can make the modification process undoable, "without backups, and additional patcher logic to restore standard settings".

**Comment by shikulja** [comment]
![image](https://github.com/user-attachments/assets/a0b66e13-f968-45fb-8a4c-69123327cea8)
The injector is fine. That's what I mean. But there may be problems with patchers.
Same as physically changing a binary file

**Comment by shikulja** [comment]
Therefore, this proposal will get rid of VanillaFixes so that one modification method is used, and without false positives (one injector is better than both, both patcher and injector)

**Comment by allfoxwy** [comment]
> Therefore, this proposal will get rid of VanillaFixes so that one modification method is used, and without false positives (one injector is better than both (both patcher and injector).

Ah I get your idea now.

Remote memory injector would trigger false alarm. And because of in early days it is the only way to build a mod. It exists. Currently people even disable their security software to use memory injector. I feel that's wrong but I won't deny injector's existence. These kind of program served us for decades and users accept it.

I think it's fine just let 2 kinds of loader on the air for now. New thing need time to be accept by people.

**Comment by shikulja** [comment]
No, I just mean to stop using the VanillaFixes patcher, need to transfer its options to .dll libraries
To use one modification method - Injector (good and flexible) compared to patcher, and it will be easier for players and cleaner for the game

**Comment by allfoxwy** [comment]
> ...to stop using the VanillaFixes patcher, need to transfer its options to .dll libraries...

VfPatcher.dll is already a .dll library, and it could be loaded by both kind of loader.

The unity of modification, it shouldn't be decided by us, user would make their choice.

**Comment by shikulja** [comment]
VfPatcher.dll and sideload-DLL do not have VanillaFixes options, or any other separate library doesn't exist either.
I don't have that choice to use VanillaFixes settings, but use injector instead of patcher

**Comment by shikulja** [comment]
oh my God, I got confused, I meant vanilla-tweaks all this time, because of this we did not understand each other

**Comment by allfoxwy** [comment]
Ah I read your point now.

I think you could open an issue report in Vanilla-tweaks git? https://github.com/brndd/vanilla-tweaks/issues

It would be best if the origin author make this change. If I implement it on my own, it would be like stealing or hijacking the project.

BTW I think currently you could patch WoW.exe with both vanilla-tweaks and sideload-DLL, they should work simultaneously. (they modify different part of the .exe file)

**Comment by shikulja** [comment]
https://github.com/brndd/vanilla-tweaks/issues/6

**Comment by allfoxwy** [comment]
Maybe create a new issue suggesting evolving into DLL?

It would be best if he come back to release a new version. He know where these patches come from so much easier to maintain them.
